test/common/runtime/runtime_impl_test.cc

```diff
@@ -678,6 +678,22 @@ TEST_F(StaticLoaderImplTest, ProtoParsing) {
   EXPECT_EQ(2, store_.gauge("runtime.num_layers", Stats::Gauge::ImportMode::NeverImport).value());
 }
 
+TEST_F(StaticLoaderImplTest, InvalidNumerator) {
+  base_ = TestUtility::parseYaml<ProtobufWkt::Struct>(R"EOF(
+    invalid_numerator:
+      numerator: 111
+      denominator: HUNDRED
+  )EOF");
+  setup();
+
+  envoy::type::v3::FractionalPercent fractional_percent;
+
+  // There is no assertion here - when numerator is invalid
+  // featureEnabled() will just drop debug log line.
+  EXPECT_CALL(generator_, random()).WillOnce(Return(500000));
+  loader_->snapshot().featureEnabled("invalid_numerator", fractional_percent);
+}
```

Comment:
Please test for the log message. You'll probably need `Envoy::LogLevelSetter save_levels(spdlog::level::debug);` before the code that triggers the message, then use `EXPECT_LOG_CONTAINS("debug", "the text", loader_->snapshot().featureEnabled("invalid_numerator", fractional_percent));`

Reply:
Thanks! :)
I didn't know that you have such type of assertion, great!